editoast/src/views/v2/timetable.rs

```diff
@@ -0,0 +1,285 @@
+use crate::decl_paginated_response;
+use crate::error::Result;
+use crate::models::List;
+use crate::models::NoParams;
+use crate::modelsv2::timetable::{Timetable, TimetableWithTrains};
+use crate::modelsv2::{Create, DeleteStatic, Model, Retrieve, Update};
+use crate::views::pagination::PaginatedResponse;
+use crate::views::pagination::PaginationQueryParam;
+use crate::DbPool;
+
+use actix_web::web::{Data, Json, Path, Query};
+use actix_web::{delete, get, post, put, HttpResponse};
+use derivative::Derivative;
+use editoast_derive::EditoastError;
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+use utoipa::ToSchema;
+
+crate::routes! {
+    "/v2/timetable" => {
+        post,
+        list,
+        "/{id}" => {
+            delete,
+            get,
+            put,
+        }
+    },
+}
+
+crate::schemas! {
+    PaginatedResponseOfTimetable,
+    TimetableForm,
+    TimetableResult,
+    TimetableDetailedResult,
+}
+
+#[derive(Debug, Error, EditoastError)]
+#[editoast_error(base_id = "timetable")]
+enum TimetableError {
+    #[error("Timetable '{timetable_id}', could not be found")]
+    #[editoast_error(status = 404)]
+    NotFound { timetable_id: i64 },
+}
+
+/// Creation form for a Timetable
+#[derive(Serialize, Deserialize, Derivative, ToSchema)]
+#[derivative(Default)]
+struct TimetableForm {
+    #[serde(default)]
+    pub electrical_profile_set_id: Option<i64>,
+}
+
+/// Creation form for a Timetable
+#[derive(Debug, Default, Serialize, Deserialize, Derivative, ToSchema)]
+struct TimetableResult {
+    pub id: i64,
+    pub electrical_profile_set_id: Option<i64>,
+}
+
+impl From<Timetable> for TimetableResult {
+    fn from(timetable: Timetable) -> Self {
+        Self {
+            id: timetable.id,
+            electrical_profile_set_id: timetable.electrical_profile_set_id,
+        }
+    }
+}
+
+/// Creation form for a Timetable
+#[derive(Debug, Default, Serialize, Deserialize, Derivative, ToSchema)]
+struct TimetableDetailedResult {
+    #[serde(flatten)]
+    #[schema(inline)]
+    pub timetable: TimetableResult,
+    pub train_ids: Vec<i64>,
+}
+
+impl From<TimetableWithTrains> for TimetableDetailedResult {
+    fn from(val: TimetableWithTrains) -> Self {
+        Self {
+            timetable: TimetableResult {
+                id: val.id,
+                electrical_profile_set_id: val.electrical_profile_set_id,
+            },
+            train_ids: val.train_ids,
+        }
+    }
+}
+
+/// Return a specific timetable with its associated schedules
+#[utoipa::path(
+    tag = "timetablev2",
+    params(
+        ("id" = u64, Path, description = "Timetable id"),
```

Comment:
Better define a `TimetableIdParam` type to factorize the field name and documentation. You can also use it in the endpoint definition `Path<TimetableIdParm>`.